src/defaults/default.js

```diff
@@ -36,36 +36,35 @@ function pipe(next, params, constants, logger) {
 const pre = cont => cont;
 
 /**
- * A standard cleanup function that takes OpenWhisk-style parameters and turns
- * them into an Express-style request object which is returned.
- * @param {Object} params Parameters following OpenWhisk convention, including
- * __ow_method and __ow_headers for HTTP requests
- * @returns {Object} A req object that is equivalent to an Express request object,
+ * A function that takes OpenWhisk-style req parameter and turns
+ * it into the original Express-style request object which is returned.
+ * @param {Object} payload Pipeline payload
+ * @param {Object} action Object representing the OpenWhisk action, including
+ * the request, __ow_method and __ow_headers for HTTP requests and secrets
+ * @returns {Object} The original req object that is equivalent to an Express request object,
  * including a headers, method, and params field
  */
-function adaptOWRequest(params) {
-  // use destructuring to drop __ow_headers and __ow_method from params
-  /* eslint-disable camelcase, no-underscore-dangle */
-  const { __ow_headers, __ow_method, ...newparams } = params;
-
-  return {
-    request: {
-      headers: __ow_headers,
-      params: newparams,
-      method: __ow_method,
-    },
-  };
-  /* eslint-enable: camelcase, no-underscore-dangle */
+function adaptOWRequest(payload, { logger, request: { params: { req = '{}' } } }) {
+  try {
+    return {
```

Comment:
Not sure I understand the question but if you refer to the fact we do not do anything with the payload, this is because the pipeline does a merge of the returned object with the current payload... whatever you return will be merged thus no need to do the merge in the function.

Btw, this is an issue for removing stuff from the payload but that's a different story.

Reply:
yes...I think we should change the pipeline from this aggregation-type to a filter-chain type.
i.e. you pass in `(payload, action)` and ignore the return value. each step is allowed to modify the objects directly. and I would rename the action to `context`, so that it is more generic.

```js
function filter(payload, action) {
  payload.response = {
     body: 'hello',
  }
}
```

alternatively, we just replace the payload/action with the returned values:

```js
function filter(payload, action) {
  ...
  return [payload, action]
}
```

Reply:
I don't like that. Modifying a function's arguments is bad enough in `pre.js`, I don't think we should make it a core pattern in the pipeline infrastructure. 

If deleting keys from the payload is an issue, I'd rather implement a filter step that removes all objects that have `undefined` value. 

With the current approach, you can easily see what you take in (because you use destructing for the parameters) and what you return (because you can just return an object literal). The alternative approach makes it much more opaque.

Reply:
I'm +1 for `context`, though.